clickhouse/tests/docker/docker-compose.yaml

```diff
@@ -0,0 +1,185 @@
+      - ./test_dictionary.xml:/etc/clickhouse-server/test_dictionary.xml
+      - ./test.csv:/opt/dictionaries/test.csv
+    depends_on:
+      - clickhouse-zookeeper
+    networks:
+      - clickhouse-network
+
+  clickhouse-02:
+    image: yandex/clickhouse-server:${CLICKHOUSE_VERSION}
+    hostname: clickhouse-02
+    container_name: clickhouse-02
+    cap_add:
+      - IPC_LOCK
+      - NET_ADMIN
+    ulimits:
+      nproc: 65536
+      nofile:
+        soft: 262144
+        hard: 262144
+    ports:
+      - "8125:8123"
+      - "9002:9000"
+    volumes:
+      - ./metrika.xml:/etc/metrika.xml
+      - ./macros/macros-02.xml:/etc/clickhouse-server/config.d/macros.xml
+      - ./users.xml:/etc/clickhouse-server/users.d/users.xml
+      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
+    depends_on:
+      - clickhouse-zookeeper
+    networks:
+      - clickhouse-network
+
+  clickhouse-03:
+    image: yandex/clickhouse-server:${CLICKHOUSE_VERSION}
+    hostname: clickhouse-03
+    container_name: clickhouse-03
+    cap_add:
+      - IPC_LOCK
+      - NET_ADMIN
+    ulimits:
+      nproc: 65536
+      nofile:
+        soft: 262144
+        hard: 262144
+    ports:
+      - "8126:8123"
+      - "9003:9000"
+    volumes:
+      - ./metrika.xml:/etc/metrika.xml
+      - ./macros/macros-03.xml:/etc/clickhouse-server/config.d/macros.xml
+      - ./users.xml:/etc/clickhouse-server/users.d/users.xml
+      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
+    depends_on:
+      - clickhouse-zookeeper
+    networks:
+      - clickhouse-network
+
+  clickhouse-04:
+    image: yandex/clickhouse-server:${CLICKHOUSE_VERSION}
+    hostname: clickhouse-04
+    container_name: clickhouse-04
+    cap_add:
+      - IPC_LOCK
+      - NET_ADMIN
+    ulimits:
+      nproc: 65536
+      nofile:
+        soft: 262144
+        hard: 262144
+    ports:
+      - "8127:8123"
+      - "9004:9000"
+    volumes:
+      - ./metrika.xml:/etc/metrika.xml
+      - ./macros/macros-04.xml:/etc/clickhouse-server/config.d/macros.xml
+      - ./users.xml:/etc/clickhouse-server/users.d/users.xml
+      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
+    depends_on:
+      - clickhouse-zookeeper
+    networks:
+      - clickhouse-network
+
+  clickhouse-05:
+    image: yandex/clickhouse-server:${CLICKHOUSE_VERSION}
+    hostname: clickhouse-05
+    container_name: clickhouse-05
+    cap_add:
+      - IPC_LOCK
+      - NET_ADMIN
+    ulimits:
+      nproc: 65536
+      nofile:
+        soft: 262144
+        hard: 262144
+    ports:
+      - "8128:8123"
+      - "9005:9000"
+    volumes:
+      - ./metrika.xml:/etc/metrika.xml
+      - ./macros/macros-05.xml:/etc/clickhouse-server/config.d/macros.xml
+      - ./users.xml:/etc/clickhouse-server/users.d/users.xml
+      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
+    depends_on:
+      - clickhouse-zookeeper
+    networks:
+      - clickhouse-network
+
+  clickhouse-06:
```

Comment:
Do we need that many replicas by default? Settling for a bit less would make sense for CI IMO.

Reply:
They all use the same pulled image version and the spin up for everything is like 18-20s, the CI time is from 6 envs